### PR TITLE
Fix/is dirty issues

### PR DESCRIPTION
**Pull request by jnation3406**

### CODE REVIEW

File: valhalla/userrequests/state_changes.py [hunk 1]
```diff
@@ -233,21 +233,22 @@ def update_request_states_for_window_expiration():
...
-        sorted_blocks_by_request = sorted(blocks, key=lambda x: x['molecules'][0]['request_number'])
-        blocks_by_request_num = {k: list(v) for k, v in itertools.groupby(sorted_blocks_by_request, key=lambda x: x['molecules'][0]['request_number'])}
-        user_request = UserRequest.objects.prefetch_related('requests').get(pk=tracking_num)
-        ur_expired = user_request.max_window_time < now
-        requests = user_request.requests.all()
-        for request in requests:
-            if request.id in blocks_by_request_num:
-                states_changed |= update_request_state(request, blocks_by_request_num[request.id], ur_expired)
-        states_changed |= update_user_request_state(user_request)
+        if tracking_num:
```
Comment (Fingel) [inline valhalla/userrequests/state_changes.py]: What does this mean? Why would we have a variable `blocks_by_tracking_num` if there was no tracking number? Better to filter out the blocks that don't have a tracking number at line 237 in the lambda so they never make it in to the variable that is named to have tracking numbers.

Comment (jnation3406) [inline valhalla/userrequests/state_changes.py]: good idea... its probably an even better idea to make the pond endpoint not return them in the first place so we don't waste the bandwidth

Comment (Fingel) [inline valhalla/userrequests/state_changes.py]: I like that even better!

File: valhalla/userrequests/state_changes.py [hunk 2]
```diff
@@ -233,14 +233,15 @@ def update_request_states_for_window_expiration():
 
 def update_request_states_from_pond_blocks(pond_blocks):
     '''Update the states of requests and user_requests given a set of recently changed pond blocks.'''
-    sorted_blocks = sorted(pond_blocks, key=lambda x: x['molecules'][0]['tracking_number'])
-    blocks_by_tracking_num = itertools.groupby(sorted_blocks, lambda x: x['molecules'][0]['tracking_number'])
+    blocks_with_tracking_nums = [pb for pb in pond_blocks if pb['molecules'][0]['tracking_num']]
```
Comment (Fingel) [inline valhalla/userrequests/state_changes.py]: Might want to use .get() here, if you aren't confident a block will have a tracking_num

**Pull request merged**